src/main/java/com/google/devtools/build/lib/remote/RemoteExecutionService.java

```diff
@@ -1049,6 +1057,20 @@ public InMemoryOutput downloadOutputs(RemoteAction action, RemoteActionResult re
           downloadsBuilder.add(downloadFile(action, file));
         }
       }
+    } else if (!regex.isEmpty()) {
+      for (FileMetadata file : metadata.files()) {
+        if (file.path.toString().matches(regex)) {
```

Comment:
I think this recompiles regex into a `Pattern` for every action. Not ideal - this should happen as a pre-process after parsing the flags. I suspect we might want to use `Predicate` here and then have a regex predicate.